lib/services/api/api_service.ts

```diff
@@ -314,4 +316,41 @@ export class SNApiService extends PureService {
     return result;
   }
 
+  async getItemRevisions(itemId: string) {
+    if (!this.session) {
+      return undefined;
+    }
+    const path = REQUEST_PATH_ITEM_REVISIONS.replace(/:item_id/, itemId);
+    const url = await this.path(path);
+    const response = await this.httpService!.getAbsolute(
+      url,
+      undefined,
+      this.session!.accessToken
+    ).catch((errorResponse: HttpResponse) => {
+      return this.errorResponseWithFallbackMessage(
+        errorResponse,
+        messages.API_MESSAGE_GENERIC_SYNC_FAIL
+      );
+    });
+    return response;
+  }
+
+  async getRevisionForItem(itemId: string, revisionId: string) {
```

Comment:
Since all revisions are returned with `getItemRevisions`, I didn't require to use this endpoint in the end.

Reply:
In this case we should modify the server to achieve the desired result. If an item with a 1MB content field has 400 revisions, `getItemRevisions` would be a 400MB request to the server, and it would probably crash the client.

We should modify /revisions endpoint on the server to only return metadata about the revision, and then use /revision/:id to download each revision when the user actually selects it.